### PR TITLE
Add pikaur support

### PR DESCRIPTION
**Pull request by GPery**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The code compiles (`cargo build`)
- [x] The code passes clippy (`cargo clippy`)
- [x] The code passes tests (`cargo test`)
- [x] The code passes rustfmt (`cargo fmt`)
- [x] *Optional:* I have tested the code myself
    - [ ] I also tested that Topgrade skips the step where needed

### CODE REVIEW

File: src/steps/os/archlinux.rs [hunk 1]
```diff
@@ -150,6 +150,40 @@ impl Pacman {
...
+    fn get() -> Option<Self> {
+        Some(Self {executable: which("pikaur")?})
+    }
+}
+
+impl ArchPackageManager for Pikaur {
+    fn upgrade(&self, ctx: &ExecutionContext) -> Result<()> {
+        let mut command = ctx.run_type().execute(&self.executable);
+
+        command
```
Comment (r-darwish) [inline src/steps/os/archlinux.rs:167]: Please add support for `--noconfirm` here as well.

**Review by r-darwish — CHANGES_REQUESTED** [review]

File: src/steps/os/archlinux.rs [hunk 2]
```diff
@@ -161,15 +195,14 @@ pub fn get_arch_package_manager(ctx: &ExecutionContext) -> Option<Box<dyn ArchPa
         config::ArchPackageManager::Autodetect => YayParu::get("paru", &pacman)
             .map(box_package_manager)
             .or_else(|| YayParu::get("yay", &pacman).map(box_package_manager))
-            .or_else(|| {
-                Trizen::get()
-                    .map(box_package_manager)
-                    .or_else(|| Pacman::get(ctx).map(box_package_manager))
-            }),
+            .or_else(|| Trizen::get().map(box_package_manager))
```
Comment (r-darwish) [inline src/steps/os/archlinux.rs:198]: This doesn't pass `cargo fmt`

**Review by r-darwish — CHANGES_REQUESTED** [review]

File: src/steps/os/archlinux.rs (hunk 1 above)
Comment (GPery) [reply src/steps/os/archlinux.rs:169]: Fixed

File: src/steps/os/archlinux.rs (hunk 2 above)
Comment (GPery) [reply src/steps/os/archlinux.rs:204]: I did forget to add changes from `cargo fmt`, but this wasn't one of them :thinking: 
Currently passes (rustfmt 1.4.38-nightly)

**Review by r-darwish — APPROVED** [review]

**Pull request merged**